core/src/main/scala/dagr/core/util/ReflectiveBuilder.scala

```diff
@@ -59,41 +60,44 @@ class ReflectiveBuilder[T](val clazz: Class[T]) {
-      val instanceMirror = if (hasCompanion) Some(mirror reflect (mirror reflectModule companionObject.get).instance) else None
-      val companionMembers = if (hasCompanion) Some(instanceMirror.get.symbol.typeSignature) else None
-
-      constructor.paramLists.head.zipWithIndex.foreach { case (param, index) =>
-        val name = param.name.toString
-        val paramType = param.typeSignature
-        if (paramType.typeArgs.size > 1) throw new CommandLineException(s"Parameter $name has multiple generic types and that is not currently supported")
-
-        val paramClass = ReflectionUtil.typeToClass(paramType)
-        val paramUnitType = if (paramType.typeArgs.size == 1) paramType.typeArgs.head else paramType
-        val paramUnitClass = ReflectionUtil.typeToClass(paramUnitType)
-        val paramPrimitiveClass = ReflectionUtil.ifPrimitiveThenWrapper(paramUnitClass)
-
-        val paramTypeName = paramUnitType match {
-          case ref: TypeRef => ref.sym.name.toString
-          case _ => paramType.typeSymbol.asClass.name.decodedName.toString
+    constructor.paramLists match {
+      case Nil | _ :: _ :: _ =>
+        throw new IllegalArgumentException("Only constructors with a single parameter list are supported.")
+      case paramList :: Nil =>
+        // More reflection objects that are needed to extract everything
+        val hasCompanion     = clazzSymbol.companion.isModule
+        val companionObject  = if (hasCompanion) Some(clazzSymbol.companion.asModule) else None
+        val instanceMirror   = companionObject map { companion => mirror reflect (mirror reflectModule companion).instance }
+        val companionMembers = instanceMirror map { mirror => mirror.symbol.typeSignature }
+
+        paramList.zipWithIndex.foreach { case (param, index) =>
+          val name = param.name.toString
+          val paramType = param.typeSignature
+          if (paramType.typeArgs.size > 1) throw new CommandLineException(s"Parameter $name has multiple generic types and that is not currently supported")
+
+          val paramClass          = ReflectionUtil.typeToClass(paramType)
+          val paramUnitType       = paramType.typeArgs.headOption getOrElse paramType
+          val paramUnitClass      = ReflectionUtil.typeToClass(paramUnitType)
+          val paramPrimitiveClass = ReflectionUtil.ifPrimitiveThenWrapper(paramUnitClass)
+
+          val paramTypeName = paramUnitType match {
+            case ref: TypeRef => ref.sym.name.toString
+            case _ => paramType.typeSymbol.asClass.name.decodedName.toString
+          }
+
+          // Sort out the default argument if one exists
+          val defaultValue: Option[_] = (companionMembers, instanceMirror) match {
+            case (None, None) => None
+            case (Some(mem), Some(mir)) =>
+              val defarg = mem.member(TermName("$lessinit$greater$default$" + (index + 1)))
+              if (defarg != NoSymbol) Option(mir.reflectMethod(defarg.asMethod)()) else None
```

Comment:
perhaps a comment that we should never hit another case, and if we do, a match exception will be thrown.